Fixed NullPointerException when formController.getInstancePath() is null

grzesiek2010 opened the pull request:
Closes #1997 

#### What has been done to verify that this works as intended?
I'm not able to reproduce the issue so I just added a null check.

#### Why is this the best possible solution? Were any other approaches considered?
It's just a null check.

#### Are there any risks to merging this code? If so, what are they?
No.

#### Do we need any specific form for testing your changes? If so, please attach one.
No.

grzesiek2010 committed:
Fixed NullPointerException when formController.getInstancePath() is null

lognaturel commented on collect_app/src/main/java/org/odk/collect/android/dao/helpers/InstancesDaoHelper.java:
```diff
@@ -44,7 +44,7 @@ public static boolean isInstanceComplete(boolean end) {
 
         FormController formController = Collect.getInstance().getFormController();
-        if (formController != null) {
+        if (formController != null && formController.getInstancePath() != null) {
```
Would be good to update the log message on line 65 as well to make it easier to track down any possible problems.

grzesiek2010 committed:
Improved naming

grzesiek2010 committed:
Improved a log

grzesiek2010 commented:
@lognaturel you are right I fixed it and I also added naming improvements replacing `instancePath` filed name with `instanceFile` since it's for Files, not Strings.

grzesiek2010 commented:
I fixed the problem visible in reports but there are a few other places where we don't check for null before to avoid crashing so maybe this is a good opportunity to review all places where `getInnstanceFile` is called. If you agree please let me do that if not you can merge.

lognaturel commented:
> there are a few other places where we don't check for null before to avoid crashing so maybe this is a good opportunity to review all places where `getInstanceFile` is called

You're absolutely right. And there are even more where `formController` methods are called without a `null` check. Last time I looked at it my assessment was that there were so many of those cases that we'd go crazy adding all the `null` checks in and that they would make the code even harder to follow. I think making improvements as we are in that code for other reasons or as crashes are actually experienced by users is a pretty good strategy. It also makes it easier to reason about each one -- sometimes the possible `null`s may be symptoms of something else bad happening. That said, if you think it makes sense to do in a sweeping pass, please do issue a PR.
